src/x/net/http/errors.go

```diff
@@ -132,7 +132,7 @@ func getStatusCode(err error) int {
 	case Error:
 		return v.Code()
 	case error:
-		if xerrors.IsInvalidParams(v) {
+		if xerrors.IsInvalidParams(v) || errors.Is(err, context.Canceled) {
```

Comment:
I believe this should be `http.StatusGatewayTimeout` since this is the same as the context deadline being exceeded, essentially a timeout.

Reply:
Hmm well StatusGatewayTimeout is 504, where this means client canceled the request. It could happen because request took too long and client canceled. For example Nginx came up with their own non 500 status code for this case https://httpstatuses.com/499

I thought smth from 4xx range would be better then 5xx in this case. Since client just abandoned the request which did not reach specified deadline.